SAMM CLI: Make native binaries easier to download via documentation (#501)
eclipse-esmf/esmf-sdk: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc

The SAMM CLI is a command line tool for the validation of Aspect Models and the generation of artifacts, such as documentation or code, from Aspect Models.

> The SAMM CLI offers multiple deployment options (you can download it bellow):

Reviewer: The SAMM CLI is available as a native executable for multiple platforms and as an an executable jar:

> The SAMM CLI is available as a native executable for multiple platforms and as an an executable jar:

Reviewer 2: ```suggestion
The SAMM CLI is available as a native executable for multiple platforms and as an executable jar:
```

Author: Yes...
Thanks!

> 

Reviewer 2: The `TIP:` and `NOTE:` are Asciidoc syntax to create these styled blocks with the icons:
![image](https://github.com/eclipse-esmf/esmf-sdk/assets/1199036/ae27b5b1-5e98-4eba-9dd0-40383e340cc4)

Using `WINDOWS:` and `LINUX:` like this will look strange. I propose we put all download links as a list in the TIP section, i.e.:

```asciidoc
[TIP] 
.Downloading samm-cli
====
* Download executable for Windows (latest): https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}-windows-x86_64.zip[samm-cli-{esmf-sdk-version}-windows-x86_64.zip]
* Download exectuable for Linux (latest): https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}-linux-x86_64.tar.gz[samm-cli-{esmf-sdk-version}-linux-x86_64.tar.gz]
* Download executable jar for other platforms and architectures (latest): https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}.jar[samm-cli-{esmf-sdk-version}.jar]. Running this version requires Java 17.
====
```

> TIP: Download latest version executable jar: icon:download[] https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}-x86_64.jar[samm-cli-{esmf-sdk-version}-x86_64.jar]

Reviewer: Download executable jar (latest):

- @atextor , @Yauhenikapl , I propose to use "native executable" instead of "native binary". Graal themselves use the wording (https://www.graalvm.org/latest/reference-manual/native-image/) and for "the basic windows user", "executable" carries "I can execute it" while "binary" can lead to ...oO(???)

NOTE: For technical reasons, Java 17 or newer is required to successfully run the tool.

> WINDOWS: Download latest version of native binaries for Windows: https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}-windows-x86_64.zip[samm-cli-{esmf-sdk-version}-windows-x86_64.zip]

Reviewer: "Download native executable for Windows (latest): "

> LINUX: Download latest version of native binaries for Linux: https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}-linux-x86_64.tar.gz[samm-cli-{esmf-sdk-version}-linux-x86_64.tar.gz]

Reviewer: "Download native executable for Linux (latest): "

The SAMM CLI is a command line tool for the validation of Aspect Models and the generation of artifacts, such as documentation or code, from Aspect Models.

The SAMM CLI is available as a native executable for multiple platforms and as an executable jar:

[TIP]
.Downloading samm-cli
====
* Download executable for Windows (latest): https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}-windows-x86_64.zip[samm-cli-{esmf-sdk-version}-windows-x86_64.zip]
* Download exectuable for Linux (latest): https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}-linux-x86_64.tar.gz[samm-cli-{esmf-sdk-version}-linux-x86_64.tar.gz]

> * Download executable jar for other platforms and architectures (latest): https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}-x86_64.jar[samm-cli-{esmf-sdk-version}-x86_64.jar]. Running this version requires Java 17.

Reviewer 2: ```suggestion
* Download executable jar for other platforms and architectures (latest): https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}.jar[samm-cli-{esmf-sdk-version}.jar]. Running this version requires Java 17.
```

----
====

> *For the Windows native binary:*

Reviewer: native executable

> . *Download*: download above or visit our github https://github.com/eclipse-esmf/esmf-sdk/releases[releases page] and download executable file for Windows.

Reviewer: *Download*: Download via above link or visit the repository of the Java SDK which contains the SAMM CLI at github https://github.com/eclipse-esmf/esmf-sdk/releases[releases page] and download the executable file for Windows.

. *Extract*: extract it to a location of your choice.
. *Open Command Prompt*: Launch the Command Prompt and navigate to the directory (samm-cli-windows) containing our application.
. *Run*: In the command prompt, execute the command ``./samm.exe` to start the application.

> *For the Linux native binary:*

Reviewer: native exectuable

. *Download*: download above or visit our github https://github.com/eclipse-esmf/esmf-sdk/releases[releases page] and download executable file for Linux.
. *Extract*: Unpack the archive to your preferred directory.
. *Open Terminal*: Open the Terminal and navigate to the directory with the application.
. *Execute*: Run the application using the command `./samm`.

> 

Reviewer: . *Input*: Make sure to read the below documentation and provide model files in the correct directory structure

((perhaps link to https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html#models-directory-structure from the same page. This is a bit of duplication, but I fear people will at that point say "got it, don't need the documentation" and then get invalid models, particularly if it's only one and not in the right folder structure. This comment should be relaxed once the SAMM CLI has been made more convenient, see https://github.com/eclipse-esmf/esmf-sdk/issues/495 ))

> Successful execution of a command is signaled by returning 0. In case of logical or some other internal problem error code 1 is returned.

Reviewer: In case of a logical or other internal error the error code 1 is being returned.

----
====

*For the Windows native executable:*

> . *Download*: Download via above link or visit the repository of the Java SDK which contains the SAMM CLI at github https://github.com/eclipse-esmf/esmf-sdk/releases[releases page] and download executable file for Windows.

Reviewer 2: ```suggestion
. *Download*: Download via above link or visit the repository of the Java SDK which contains the SAMM CLI at the Github https://github.com/eclipse-esmf/esmf-sdk/releases[releases page] and download executable file for Windows.
```

. *Extract*: extract it to a location of your choice.
. *Open Command Prompt*: Launch the Command Prompt and navigate to the directory (samm-cli-windows) containing our application.

> . *Input*: Make sure to read the below documentation and provide model files in the correct models directory structure[directory structure].

Reviewer 2: ```suggestion
. *Input*: Make sure to read the below documentation and provide model files in the correct xref:models-directory-structure[directory structure].
```

> . *Run*: In the command prompt, execute the command ``./samm.exe` to start the application.

Reviewer 2: ```suggestion
. *Run*: In the command prompt, execute the command `samm` to start the application. See the below table for details.
```

*For the Linux native executable:*

> . *Download*: download above or visit our github https://github.com/eclipse-esmf/esmf-sdk/releases[releases page] and download executable file for Linux.

Reviewer 2: ```suggestion
. *Download*: Download via above link or visit the repository of the Java SDK which contains the SAMM CLI at the Github https://github.com/eclipse-esmf/esmf-sdk/releases[releases page] and download executable file for Linux.
```

. *Extract*: Unpack the archive to your preferred directory.
. *Open Terminal*: Open the Terminal and navigate to the directory with the application.

> . *Input*: Make sure to read the below documentation and provide model files in the correct models directory structure[directory structure].

Reviewer 2: ```suggestion
. *Input*: Make sure to read the below documentation and provide model files in the correct xref:models-directory-structure[directory structure].
```
